apps/accelerate/chatllama/chatllama/rlhf/trainer.py

```diff
@@ -263,18 +296,6 @@ def __init__(
         if not os.path.exists(self.config.trainer.checkpoint_folder):
             os.mkdir(self.config.trainer.checkpoint_folder)
 
-        self.model_engine = None  # deepspeed model engine
-        if self.deepspeed_enable is True:
-            if self.deepspeed_config_path is None:
-                raise ValueError(
-                    "DeepSpeed config path is None, but deepspeed is enabled"
-                )
-            if os.path.exists(self.deepspeed_config_path) is False:
-                raise ValueError(
-                    f"DeepSpeed config path {self.deepspeed_config_path}"
-                    f"does not exist"
-                )
-
```

Comment:
No DeepSpeed for RLHF?

Reply:
ok the problem is the following... 
In vanilla deepspeed for what i understood the flow is the following...
(
    self.model_engine,
    self.optimizer,
    self.train_dataloader,
    _,
) = deepspeed.initialize(
    args=None,
    model=self.model,
    model_parameters=self.model.parameters(),
    training_data=self.train_dataloader,
    config=self.config.deepspeed_config_path,
)

then 
output = self.model_engine(input, mask)
loss = ...
self.model_engine.backward(loss)
self.model_engine.step()

When i train with the RLHF i have a class ActorCritic which contains two models actor and critic.
So the forward is performed as self.actorcritic.forward()
and thus i will need to do a model_engine of actorcritic.
Then the update is done separately with two different losses 
loss_actor = ... 
loss_critic = ... 
and first is updated only the actor and then only the critic so i should have a model engine for actor and critic... 
TO apply deepspeed in this case we have to rethink some classes and how are structured, maybe is best that we first try to use it vanilla with the other training and see what works and what not and then try to refactor this things to make it work also for RLHF.